app/src/org/commcare/activities/connect/ConnectIDNetworkHelper.java

```diff
@@ -0,0 +1,267 @@
+package org.commcare.activities.connect;
+
+import android.content.Context;
+import android.widget.Toast;
+
+import com.google.common.collect.ArrayListMultimap;
+import com.google.common.collect.ImmutableMultimap;
+import com.google.common.collect.Multimap;
+import com.google.gson.Gson;
+
+import org.commcare.core.interfaces.HttpResponseProcessor;
+import org.commcare.core.network.AuthInfo;
+import org.commcare.core.network.HTTPMethod;
+import org.commcare.core.network.ModernHttpRequester;
+import org.commcare.interfaces.ConnectorWithHttpResponseProcessor;
+import org.commcare.tasks.ModernHttpTask;
+import org.commcare.tasks.templates.CommCareTask;
+import org.javarosa.core.io.StreamsUtil;
+import org.javarosa.core.services.Logger;
+import org.javarosa.core.services.locale.Localization;
+import org.json.JSONException;
+import org.json.JSONObject;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.HashMap;
+import java.util.Map;
+
+import okhttp3.MediaType;
+import okhttp3.RequestBody;
+
+public class ConnectIDNetworkHelper {
+    public interface INetworkResultHandler {
+        void processSuccess(int responseCode, InputStream responseData);
+        void processFailure(int responseCode, IOException e);
+    }
+
+    public static void post(Context context, String url, AuthInfo authInfo, HashMap<String, String> params, INetworkResultHandler handler) {
+        Gson gson = new Gson();
+        String json = gson.toJson(params);
+
+        RequestBody requestBody = RequestBody.create(MediaType.parse("application/json"), json);
+        ModernHttpTask postTask =
```

Comment:
(Optional)There is potentially an opportunity here to experiment with Kotlin Coroutines instead of relying on async task framework. Async tasks are deprecated and we want to move away from them and Connect stuff is potentially idle to implement that first without introducing regressions in the main CC code.